how-to/workspace-platform-starter/client/src/framework/shapes/favorite-shapes.ts

```diff
@@ -0,0 +1,217 @@
+import type { PlatformStorageMetadata } from "./platform-shapes";
+
+/**
+ * Names for all the favorite types.
+ */
+export type FavoriteTypeNames = "app" | "workspace" | "page" | "query";
+
+/**
+ * Options for the favorite provider.
+ */
+export interface FavoriteProviderOptions {
+	/**
+	 * The icon that should be used if you want to indicate this is a favorite action
+	 */
+	favoriteIcon: string;
+
+	/** The icon to use to indicate that this favorite can be unset */
+	unfavoriteIcon: string;
+
+	/**
+	 * What commands should integrations check for if they intent to support the display of favorites
+	 */
+	favoriteCommand?: string;
+
+	/**
+	 * The connection provider can have actions registered against it from the platform. This provides a default list of
+	 * actions that connections should be able to use if actions are enabled for that connection.
+	 */
+	supportedFavoriteTypes?: FavoriteTypeNames[];
+}
+
+/**
+ * When an entry is made it represents a type supported by this platform. This can be used to lookup and launch the thing this entry refers to.
+ */
+export interface FavoriteEntry {
+	/**
+	 * A unique guid to represent this favorite entry so that it can be updated or removed
+	 */
+	id: string;
+
+	/**
+	 * The id for the favorite type this entry represents
+	 */
+	typeId: string;
+
+	/**
+	 * What type of favorite entry does this entry represent
+	 */
+	type: FavoriteTypeNames;
+
+	/**
+	 * The timestamp for the entry.
+	 */
+	timestamp: Date;
+
+	/**
+	 * Does this favorite have a suggested label that can be used to avoid a lookup
+	 */
+	label?: string;
+
+	/**
+	 * Does this favorite have a suggested icon that can be used to avoid a lookup
+	 */
+	icon?: string;
+}
+
+/**
+ * Info to return to interested parties to help them support favorites
+ */
+export interface FavoriteInfo {
+	/**
+	 * The path to an icon that can be used to indicate the ability to favorite
+	 */
+	favoriteIcon?: string;
+	/**
+	 * The path to an icon that can be used to indicate the ability to remove this favorite
+	 */
+	unfavoriteIcon?: string;
+	/**
+	 * A command that supporting modules should listen for if they are to display favorites that fall under them
+	 */
+	command?: string;
+	/**
+	 * What types of favorite item are supported on this platform
+	 */
+	enabledTypes?: FavoriteTypeNames[];
+	/**
+	 * Is favorite support enabled on this platform.
+	 */
+	isEnabled: boolean;
+}
+
+/**
+ * A client that can be used to provide access to some or all of the favorite functionality
+ */
+export interface FavoriteClient {
+	/**
+	 * The ability to request supporting information about whether favorites are initialized for the platform and supporting information.
+	 * @returns Supporting information.
+	 */
+	getInfo: () => FavoriteInfo;
```

Comment:
The method definitions use arrow expressions, they should just be plain functions
e.g. `getInfo(): FavoriteInfo;`

Reply:
done